### PR TITLE
E0d/devstack nightly

### PR DESCRIPTION
**Pull request by e0d**
Changes to support nightly builds of DevStack

### CODE REVIEW

**Comment by e0d** [comment]
@singingwolfboy @feanil 

Opening to spur discussion.  I'd like to move this forward so that nightly builds of devstack are an option for edx developers.

File: vagrant/base/devstack/Vagrantfile [hunk 1]
```diff
@@ -49,6 +49,19 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     config.ssh.forward_x11 = true
   end
 
+  # 
+  # By default no shared filesystem is configured between the guest and host.
+  # To enable filesystem sharing, which makes editing running code posible,
+  # you will need to enable either NFS of VBOXFS.
```
Comment (singingwolfboy) [inline vagrant/base/devstack/Vagrantfile:65]: Why is this change necessary? It's going to need a lot of documentation and announcements.

Comment (feanil) [inline vagrant/base/devstack/Vagrantfile:65]: This is only in the base devstack not on the release version that developers use.  I assume this is to make it easier to build these boxes @e0d ?

**Comment by feanil** [comment]
:+1:

File: vagrant/base/devstack/Vagrantfile (hunk 1 above)
Comment (carsongee) [inline vagrant/base/devstack/Vagrantfile:65]: Why not just add a third option, like `VAGRANT_NO_SHARE` so that this isn't backwards incompatible?

Comment (e0d) [inline vagrant/base/devstack/Vagrantfile:65]: I considered doing that way, this felt cleaner, but I'm happy to go the other way if that's majority opinion.  I'm not sure who all the customers of the base scripts are.  But not need to allow jenkins to sudo to mount NFS shares during builds seemed preferable to me.

This shouldn't prevent the release from automatically creating the mounts.

Comment (carsongee) [inline vagrant/base/devstack/Vagrantfile:65]: I suppose I'm probably the primary user then, since I develop for both config and edx-platform dev it is convenient to use the one box. If I'm the only one, feel free to keep it as written, I can manage my environment variables.

Comment (e0d) [inline vagrant/base/devstack/Vagrantfile:65]: I'm happy to fence it off in that case.
